pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -550,24 +558,39 @@ func reconcileClusterRoleBinding(ctx context.Context, cl client.Client, c *green
 }
 
 // reconcileRoleBinding creates or updates a RoleBinding in the Cluster the given client.Client is created for
-func reconcileRoleBinding(ctx context.Context, cl client.Client, c *greenhousev1alpha1.Cluster, rb *rbacv1.RoleBinding) error {
+func reconcileRoleBinding(ctx context.Context, cl client.Client, c *greenhousev1alpha1.Cluster, rb *rbacv1.RoleBinding, createNamespaces bool) error {
 	remoteRB := &rbacv1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      rb.Name,
 			Namespace: rb.Namespace,
 		},
 	}
 
+	if createNamespaces {
+		namespace := new(corev1.Namespace)
+		err := cl.Get(ctx, types.NamespacedName{Name: rb.Namespace}, namespace)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				err := cl.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rb.Namespace}})
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+	}
+
 	result, err := clientutil.CreateOrPatch(ctx, cl, remoteRB, func() error {
 		remoteRB.Labels = rb.Labels
 		remoteRB.RoleRef = rb.RoleRef
 		remoteRB.Subjects = rb.Subjects
 		return nil
 	})
-
 	if err != nil {
 		return err
 	}
```

Comment:
This is what I meant before. If we move the check for the `apierrors.IsNotFound` into the error handling for the CreateOrPatch, then it will fail the first reconciliation if the namespace does not exist. But since if the createNamespace is true, then it will create the namespace during error handling and succeed on the next reconciliation.
Doing it this way there is no need to do the `Get(..)` on the namespace on every iteration.  
```suggestion
	if err != nil {
		if apierrors.IsNotFound(err) && createNamespaces {
			err := cl.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rb.Namespace}})
			if err != nil {
				return err
			}
			return errors.New("failed to create RoleBinding, namespace did not exist")
		}
	return err
	}
```